src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -479,6 +530,52 @@ public String getRepositoryUrl() {
         }
     }
 
+    /**
+     * Sets all fields on this source appropriately for the provided URL.
+     *
+     * @param repositoryUrl HTTP URL for the repository to connect to
+     * @since 2.6.0
+     */
+    @DataBoundSetter
+    public void setRepositoryUrl(@CheckForNull String repositoryUrl)  {
+        GitHubRepositoryInfo info = GitHubRepositoryInfo.forRepositoryUrl(repositoryUrl);
+        this.apiUri = info.getApiUri();
+        this.repoOwner = info.getRepoOwner();
+        this.repository = info.getRepository();
+        this.repositoryUrl = info.getRepositoryUrl();
+    }
+
+    /**
+     * This method is called after Stapler instantiates this class from the Jenkins UI.
+     * This allows the object react to parameters in the UI without making them part of the
+     * data in the class.  It also lets the class have more complex behavior that would
+     * be allowed by data binding alone.
+     *
+     * We look at configuredByUrl to decide which set of parameters we look at to
+     * configure this instance.  This method is not called when instantiating from
+     * other scenarios such as JobDSL, so the object still need to have a reasonable
+     * way of handling callers setting multiple incompatible properties.
+     *
+     * @param staplerRequest the full request sent from the UI
+     * @param json the raw data of the request
+     * @return new GitHubSCMSource with settings matching provided JSON
+     */
+    @Override
+    public GitHubSCMSource bindResolve(StaplerRequest staplerRequest, JSONObject json) {
+        boolean configuredByUrl = json.getBoolean("configuredByUrl");
+
+        if (configuredByUrl) {
+            this.setRepositoryUrl(json.get("repositoryUrl").toString());
+        } else {
+            this.repositoryUrl = null;
+            this.apiUri = json.get("apiUri").toString();
```

Comment:
```suggestion
            this.apiUri = setApiUri(json.get("apiUri").toString());
```
Otherwise we never go through `normalizeApiUri` on this path.